src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -435,52 +462,60 @@ public GenomicsDBImporter(Map<String, FeatureReader<VariantContext>> sampleToVCM
+                            ChromosomeInterval chromosomeInterval,
+                            String workspace,
+                            String arrayname,
+                            Long vcfBufferSizePerColumnPartition,
+                            Long segmentSize,
+                            Long lbRowIdx,
+                            Long ubRowIdx,
+                            boolean useSamplesInOrderProvided,
+                            boolean failIfUpdating,
+                            int rank) throws IOException {
     // Mark this flag so that protocol buffer based vid
     // and callset map are propagated to C++ GenomicsDBImporter
     mUsingVidMappingProtoBuf = true;
 
     GenomicsDBImportConfiguration.ImportConfiguration importConfiguration =
-      createImportConfiguration(workspace, arrayname, vcfBufferSizePerColumnPartition, segmentSize, failIfUpdating);
-    File importJSONFile = printLoaderJSONFile(importConfiguration, "");
+      createImportConfiguration(workspace, arrayname, vcfBufferSizePerColumnPartition,
+                                segmentSize, failIfUpdating);
+    File importJSONFile = dumpTemporaryLoaderJSONFile(importConfiguration, "");
 
-    mVidMap = generateVidMapFromMergedHeader(mergedHeader);
-    mCallsetMap = generateSortedCallSetMap(sampleToVCMap, useSamplesInOrderProvided, lbRowIdx);
+    GenomicsDBVidMapProto.VidMappingPB vidMapPB = generateVidMapFromMergedHeader(mergedHeader);
+    GenomicsDBCallsetsMapProto.CallsetMappingPB callsetMapPB =
+        generateSortedCallSetMap(sampleToReaderMap, useSamplesInOrderProvided, lbRowIdx);
 
     initialize(importJSONFile.getAbsolutePath(), rank, lbRowIdx, ubRowIdx);
 
     mGenomicsDBImporterObjectHandle =
-      jniInitializeGenomicsDBImporterObject(mLoaderJSONFile, mRank, mLbRowIdx, mUbRowIdx);
+      jniInitializeGenomicsDBImporterObject(mLoaderJSONFile, mRank, lbRowIdx, ubRowIdx);
+
+    jniCopyVidMap(mGenomicsDBImporterObjectHandle, vidMapPB.toByteArray());
+    jniCopyCallsetMap(mGenomicsDBImporterObjectHandle, callsetMapPB.toByteArray());
+
+    for (GenomicsDBCallsetsMapProto.SampleIDToTileDBIDMap sampleToIDMap :
+        callsetMapPB.getCallsetsList()) {
 
-    jniCopyVidMap(mGenomicsDBImporterObjectHandle, mVidMap.toByteArray());
-    jniCopyCallsetMap(mGenomicsDBImporterObjectHandle, mCallsetMap.toByteArray());
+      String sampleName = sampleToIDMap.getSampleName();
 
-    for (GenomicsDBCallsetsMapProto.SampleIDToTileDBIDMap sampleInfo :
-      mCallsetMap.getCallsetsList()) {
-        FeatureReader<VariantContext> featureReader =
-          sampleToVCMap.get(sampleInfo.getSampleName());
-        CloseableIterator<VariantContext> iterator =
-          featureReader.query(chromosomeInterval.getContig(),
+      FeatureReader<VariantContext> featureReader = sampleToReaderMap.get(sampleName);
+
+      if (featureReader==null) {
+        throw new IllegalArgumentException("Null FeatureReader found for sample: " + sampleName);
```

Comment:
You probably want to move this check closer to the top of method so it fails fast instead of in the middle of a batch import.